### PR TITLE
Java string preprocessing: Remove or document unused parameter function_id

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: d29be11e86ab4b67f695bfa0a4c3e75e770de175).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77138015

**Review by kroening — APPROVED** [review]

**Comment by smowton** [comment]
Why remove some and not others?

**Comment by tautschnig** [comment]
> Why remove some and not others?

I'm working towards making it more complete.

**Review by peterschrammel — DISMISSED** [review]
Removing this is wrong. This parameter is necessary and must be used at https://github.com/diffblue/cbmc/blob/develop/jbmc/src/java_bytecode/java_string_library_preprocess.cpp#L662.
Besides that, I noticed that there are several places where `function_id` must be used as second argument of `fresh_aux_symbol`, but it isn't.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 1275ca29e7638e3eda6dc6e2c41c6ff3fda4a2d0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77172942

**Comment by tautschnig** [comment]
@peterschrammel @cesaro and anyone else with sufficient knowledge of the matter: I accept that this change may not be the right one, but really just an unused parameter is getting removed (and both CBMC and TG apparently continue to build fine). I'm afraid I do not have sufficient knowledge on how the parameter should be used instead, thus re-assigning.

**Comment by peterschrammel** [comment]
This  goes into the right direction. Some more work will need to be done as a follow up, but is not urgent: https://github.com/diffblue/cbmc/issues/2555

**Comment by peterschrammel** [comment]
@tautschnig, there are compilation errors: https://travis-ci.org/diffblue/cbmc/jobs/401223946#L1929

**Comment by tautschnig** [comment]
@peterschrammel The syntax errors have been fixed.

File: jbmc/src/java_bytecode/java_string_library_preprocess.cpp [hunk 1]
```diff
@@ -1832,11 +1847,17 @@ codet java_string_library_preprocesst::make_string_length_code(
...
+  (void)function_id;
+
   code_typet::parameterst params=type.parameters();
   symbol_exprt arg_this(params[0].get_identifier(), params[0].type());
   dereference_exprt deref=
     checked_dereference(arg_this, arg_this.type().subtype());
-  return code_returnt(get_length(deref, symbol_table));
+
+  code_returnt ret(get_length(deref, symbol_table));
+  ret.add_source_location() = loc;
```
Comment (allredj) [inline jbmc/src/java_bytecode/java_string_library_preprocess.cpp]: Nit: should be in separate commit

**Review by allredj — COMMENTED** [review]
The commit message no longer corresponds to the content of the commit.

**Review by smowton — APPROVED** [review]
change PR title to make clear there are other unrelated cleanups in here too; otherwise lgtm

File: jbmc/src/java_bytecode/java_string_library_preprocess.cpp (hunk 1 above)
Comment (tautschnig) [reply jbmc/src/java_bytecode/java_string_library_preprocess.cpp]: Moved to PR #2726.
